### ISSUE TITLE
New tab page sponsored image (NTP SI) P3A metrics for users not opted in to rewards

### ISSUE THREAD

**Issue by DJAndries** [brave/brave-browser#24609]
P3A metrics will be dynamically created for each NTP-SI campaign. Views, clicks and landings will be recorded for each campaign.

The name format will be as follows: `creativeInstanceId.{campaign id}.{event type}`.

These metrics will be reported daily, for up to seven days per campaign. When the metrics are uploaded, the header `X-Brave-P3A-Creative` will be set to `true`.

This issue depends on #24318 and #24319 .

**Comment by GeetaSarvadnya** [comment]
@DJAndries I would need a clarification on step 4 and 5. In the step 4, I have adavanced the sys clock by one day (i.e 3rd Nov) when I relaunch the browser, I don't see `creativeInstanceId` in the brave://local-state file, Is that expected? Also, I could see the header `P3A-creative.brave.com` listed in the charles proxy network monitoring tool as below. In this case how do we know the metrics are sent with the flag `true`? the step 5 also behaves same as step 4. What is the difference between the step 4 and step 5?
![image (15)](https://user-images.githubusercontent.com/38657976/199451996-dd0ef381-9544-46b3-b0be-6d2be3043658.png)

**Comment by DJAndries** [comment]
@GeetaSarvadnya Once the metric has been sent, you should not see the `creativeInstanceId`-prefixed metric on the local state page (unless you viewed/clicked/landed on the same ad between the time the value was recorded by the P3A system and the time the metric was sent). The header is no longer relevant, only the URL `https://p3a-creative.brave.com`. I updated the test plan accordingly. In step 5, you should ensure that the metric sent in step 4 is not resent, unless you viewed/clicked/landed on the same ad again.

**Comment by GeetaSarvadnya** [comment]
Removing the QA Pass label as this issue will be uplifted to 1.45 via https://github.com/brave/brave-core/pull/15745 and testing will be done in 1.45.x

**Comment by kjozwiak** [comment]
The above requires `1.45.121` or higher for `1.45.x` verification 👍

**Comment by GeetaSarvadnya** [comment]
Verification is inprogress


```
Brave | 1.45.121 Chromium: 107.0.5304.91 (Official Build) (64-bit)
-- | --
Revision | 3d5948960d62418160796d5831a4d2d7d6c90fa8-refs/branch-heads/5304@{#1097}
OS | Windows 10 Version 21H2 (Build 19044.2130)

```

**Comment by GeetaSarvadnya** [comment]
@DJAndries I don't see landed creativeinstanceId in brave://local-state even which I click on NTP SI and stay on the landing page for more than 10 secs. This was working fine when I tested the issue in `1.46.81`. what could be the reason, any idea?

**Comment by GeetaSarvadnya** [comment]
Landed creativeinstanceId's are shown now, thanks @DJAndries !

**Comment by GeetaSarvadnya** [comment]
Verification is inprogress on `Vivo X70 Pro version 12` running `Bravemonoarm64.apk_1.45.121`

- Verified the test plan from https://github.com/brave/brave-core/pull/14595

<details>
<summary>Verify creativeInstanceId when ads rewards is disabled_</summary>

- Confirmed that `creativeInstanceId` doesn't exists in brave://local-state when ads and rewards is disabled


</details>

<details>
<summary>Verify creative metric with the suffix views_</summary>

- Clean profile 1.46.x
- Open brave://local-state file
- Open NTP until NTP SI is shown
- Reload the brave://local-state file 
- Confirmed that one creative metric with the suffix `views` exists in the local state page

Example | Example
----------|----------



</details>

<details>
<summary>Verify creative metric with the suffix clicks_</summary>

- Clean profile 1.46.x
- Open brave://local-state file
- Open NTP until NTP SI is shown
- Reload the brave://local-state file 
- Confirmed that one creative metric with the suffix `views` exists in the local state page
- Click on the creative (NTP SI) 
- Confirmed that one creative metric with the suffix `clicks` exists in the local state page

Example | Example | Example
----------|----------|----------



</details>


<details>
<summary>Verify creative metrics are sent to the endpoint https://p3a-creative.brave.com_</summary>

- Clean profile 1.46.x
- Open brave://local-state file
- Open NTP until NTP SI is shown
- Reload the brave://local-state file 
- Confirmed that one creative metric with the suffix `views` exists in the local state page
- Click on the creative (NTP SI) and stay on the landing page for 10 secs
- Confirmed that one creative metric with the suffix `clicks` and `lands` exists in the local state page

Example | Example | Example
----------|----------|----------




- Close the browser
- Set the sys date ahead by one day
- Open charles proxy
- Relaunch the browser and ensured creative metrics are sent to the endpoint https://p3a-creative.brave.com

View | Click | Land
-----|-------|-------





- Open brave://local-state file and ensured creative metrics are NOT shown in local-state file
![image](https://user-images.githubusercontent.com/38657976/199654332-774f091e-5359-4916-901b-513063fb9398.png)

- Close browser, advance the time by one day, open browser, ensure the metrics are not resent.

brave://local-state | https://p3a-creative.brave.com
--------------------|-----


</details>
